account_invoice_inter_company/models/account_invoice.py

```diff
@@ -0,0 +1,282 @@
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import AccessError, UserError
+from odoo.tools import float_compare
+
+
+class AccountInvoice(models.Model):
+
+    _inherit = 'account.invoice'
+
+    auto_generated = fields.Boolean(
+        'Auto Generated Document',
+        copy=False, default=False)
+    auto_invoice_id = fields.Many2one(
+        'account.invoice',
+        string='Source Invoice',
+        readonly=True, copy=False,
+        prefetch=False)
+
+    @api.multi
+    def _find_company_from_invoice_partner(self):
+        self.ensure_one()
+        company = self.env['res.company'].sudo().search([
+            ('partner_id', '=', self.partner_id.id)
+        ], limit=1)
+        return company or False
+
+    @api.multi
+    def action_invoice_open(self):
+        """ Validated invoice generate cross invoice base on company rules """
+        res = super(AccountInvoice, self).action_invoice_open()
+        for src_invoice in self:
+            # do not consider invoices that have already been auto-generated,
+            # nor the invoices that were already validated in the past
+            dest_company = src_invoice._find_company_from_invoice_partner()
+            if not dest_company or src_invoice.auto_generated:
+                return res
```

Comment:
s/return res/continue/g  use continue instead of return to allow managing multiple invoice validation (case when self contains inter company and simple invoice)

Reply:
@astirpe what do you think about @mourad-ehm proposal `continue` instead of `return res`

I think it's a good thing.

Thanks

Reply:
Yes indeed, must be continue. Thanks! :+1: